lambdas/service/app.py

```diff
@@ -488,9 +490,16 @@ def static_resource(file):
     cache_control='no-store'
 )
 def oauth2_redirect():
-    oauth2_redirect_html = app.load_static_resource('swagger', 'oauth2-redirect.html')
+    oauth2_redirect_template = app.load_static_resource('swagger',
+                                                        'oauth2-redirect.html.template.mustache')
+    nonce = secrets.token_urlsafe(32)
+    oauth2_redirect_html = chevron.render(oauth2_redirect_template, {
+        'NONCE': json.dumps(nonce)
+    })
+    # The nonce value is passed via the x-azul-nonce header so that it can be
+    # processed by the middleware handler.
     return Response(status_code=200,
-                    headers={"Content-Type": "text/html"},
+                    headers={'Content-Type': 'text/html', 'x-azul-nonce': nonce},
```

Comment:
This is what confused me last time. Why do you need to take the detour via a custom header? Couldn't you just inject the final header here?